botocore/client.py

```diff
@@ -666,6 +677,7 @@ def __init__(self, events, client_config, endpoint_url, service_model,
         self._endpoint_url = endpoint_url
         self._service_model = service_model
         self._method_to_api_mapping = method_to_api_mapping
+        self._error_factory = ServiceErrorFactory(self._service_model)
```

Comment:
Due to the amount of indirection between session and client, I'm not clear on how to scope an exception factory to a session. Can you show me?